### PR TITLE
fix(condo): DOMA-7735, DOMA-7702, DOMA-7730, DOMA-7739, DOMA-6424: fixed meter reporting period, property meter readings and accesses for bulk operations

### PR DESCRIPTION
**Pull request by Alllex202**
- [x] fix(condo): DOMA-7735 fixed employee access for bulk operations (An employee could manage objects without access to them if he send them along with objects to which he has access)

 - [x] fix(condo): DOMA-7702 fixed deleting of meter reporting period (It was impossible to delete property meters due to a ban on bulk operations)

 - [x] fix(condo): DOMA-7730 fixed updating of meter reporting period (It was impossible to remove the flag for selecting all properties)

 - [x] fix(condo): DOMA-7739 added check that the property is from the specified organization (MeterReportingPeriod)

 - [x] fix(condo): DOMA-6424 fixed deleting of property meter readings (when deleting a proprty meter, the property meter readings were not deleted)

### CODE REVIEW

File: apps/condo/domains/meter/access/MeterReportingPeriod.js [hunk 1]
```diff
@@ -42,44 +49,61 @@ async function canReadMeterReportingPeriods ({ authentication: { item: user } })
...
+    const isBulkRequest = Array.isArray(originalInput)
 
-            if (!property) return false
+    let organizationIds = []
 
-            propertyOrganization = get(property, 'organization', null)
+    if (operation === 'create') {
+        if (isBulkRequest) {
+            return false
+        } else {
```
Comment (sitozzz) [inline apps/condo/domains/meter/access/MeterReportingPeriod.js:64]: Why you can't use bulk create, but can use bulk update?

File: apps/condo/domains/meter/schema/MeterReportingPeriod.js [hunk 2]
```diff
@@ -59,16 +64,31 @@ const MeterReportingPeriod = new GQLListSchema('MeterReportingPeriod', {
...
             knexOptions: { isNotNullable: false },
             kmigratorOptions: { null: true, on_delete: 'models.CASCADE' },
+            hooks: {
+                validateInput: async ({ resolvedData, fieldPath, existingItem, context }) => {
+                    const newItem = { ...existingItem, ...resolvedData }
+                    const propertyId = newItem[fieldPath]
+                    const organizationId = newItem.organization
+
+                    if (propertyId) {
+                        const property = await getById('Property', propertyId)
```
Comment (sitozzz) [inline apps/condo/domains/meter/schema/MeterReportingPeriod.js:79]: deletedAt?

File: apps/condo/domains/organization/utils/accessSchema.js [hunk 3]
```diff
@@ -60,44 +44,84 @@ async function checkRelatedOrganizationPermission (userId, organizationId, permi
...
  */
 async function checkPermissionsInUserOrganizationsOrRelatedOrganizations (userId, organizationIds, permission) {
-    if (!userId || !organizationIds.length) return false
-    const hasPermissionInRelatedOrganizations = await checkRelatedOrganizationsPermission(userId, organizationIds, permission)
-    const hasPermissionInUserOrganizations = await checkOrganizationsPermission(userId, organizationIds, permission)
+    if (!userId || !isArray(organizationIds) || isEmpty(organizationIds)) return false
+
+    const uniqOrganizationIds = uniq(organizationIds)
+
+    if (uniqOrganizationIds.some(orgId => !isString(orgId))) return false
```
Comment (sitozzz) [inline apps/condo/domains/organization/utils/accessSchema.js:117]: What the real reason of this check? Correct me if I'm wrong, but are you trying to check if one of the input of organizationIds is not a valid uuid string? If that true, you should add a proper check for uuid pattern via regex or uuid validation utils

File: apps/condo/domains/meter/access/MeterReportingPeriod.js (hunk 1 above)
Comment (Alllex202) [reply apps/condo/domains/meter/access/MeterReportingPeriod.js:64]: I don't think there is a need for this right now

File: apps/condo/domains/organization/utils/accessSchema.js (hunk 3 above)
Comment (Alllex202) [reply apps/condo/domains/organization/utils/accessSchema.js:117]: The idea was to immediately block requests with an empty organizations. But yes, we can immediately check the correctness of the identifier, I’ll add this

**Review by VKislov — APPROVED** [review]

File: apps/condo/domains/meter/schema/MeterReportingPeriod.js (hunk 2 above)
Comment (Alllex202) [reply apps/condo/domains/meter/schema/MeterReportingPeriod.js]: fixed

File: apps/condo/domains/organization/utils/accessSchema.js (hunk 3 above)
Comment (Alllex202) [reply apps/condo/domains/organization/utils/accessSchema.js]: Done

**Review by sitozzz — APPROVED** [review]

**Pull request merged**